lib/spack/docs/developer_guide.rst

```diff
@@ -861,6 +824,39 @@ We merge each release into ``develop``, we tag the latest release as
    The ``--force`` argument makes ``git`` overwrite the existing
    ``releases/latest`` tag with the new one.
 
+We also merge each new latest release into ``develop``. Make sure that
+it is merged back into ``develop`` with a merge commit:
+
+.. code-block:: console
+
+   $ git checkout develop
+   $ git merge --no-ff releases/vX.Y  # vX.Y is the new release's branch
+   $ git push
+
```

Comment:
This might be too detailed but I think it's important.  Suggest adding:

```
Note that we merge the *release branch* and not the `releases/latest` tag,
because merging the release branch produces a more precise commit message that
includes the release version.
```

By this logic, though, we should merge the release tag, as that is even more precise, e.g.:

```
   $ git checkout develop
   $ git merge --no-ff vX.Y.Z  # the new release's tag
   $ git push
```

This will make a default commit message that says "merged v0.15.1 into develop" instead of "merged releases/v0.15", which IMO is a little better.

Any thoughts on this?  You basically have 3 choices: merge `releases/latest`, `releases/vX.Y`, or `vX.Y.Z`.  They're mostly equivalent aside from the commit message.

Reply:
I like the idea of merging the tag. I'll update that. I don't think we need the longer caveat though (and the section is quite "explainey" already.